runbooks, Add info about KubeMacPoolDuplicateMacsFound runbook (#23)
kubevirt/monitoring: docs/runbooks/KubeMacPoolDuplicateMacsFound.md

# KubeMacPoolDuplicateMacsFound

## Meaning

> KubeMacPool is responsible to allocate mac addresses and avoid mac address conflicts.

Reviewer: ```suggestion
KubeMacPool is responsible to allocate MAC Addresses and prevent MAC Address conflicts.
```

> Upon starting KubeMacPool, it scan the current VM instances and updates it's database.

Reviewer: ```suggestion
Upon starting KubeMacPool, it scans the cluster for MAC Addresses in VMs set on the managed namespaces.
```

> In case it find duplicates it increase a counter of mac duplicate, which results in this alert.

Reviewer: ```suggestion
In case it finds duplicates it will raise this alert.
```

> Duplicates will not be updated in the database but do exists until solved.

Reviewer: This is not accurate, I would vote to remove this line.. we should keep the explanation generic and simple
```suggestion
```

## Impact

> Having a duplicate mac address can break the requirement that each interface on the same LAN,

Reviewer: ```suggestion
Having a duplicate MAC Addresses can break the requirement that each interface on the same LAN,
```

> should have a unique mac address.

Reviewer: ```suggestion
should have a unique MAC Address.
```

## Diagnosis

> - Check the logs of kubemacpool-mac-controller, find the lines that say

Reviewer: I would be more specific.. don't expect the sys-admin to know what's KMP's namespace
```
- Get kubemacpool-mac-controller pod namespace + command example
- Extract the conflicting VMs from kubemacpool pod logs, from which you will have 1, 2 and 3: + command example
```

  `already allocated ... already allocated`
  You can use this commands:
  `oc logs -n <kubemacpool-namespace> <kubemacpool-mac-controller-pod-name> | grep already`
  Those lines have all the information that is needed:
    - The mac address which has a conflict
    - The current entry in the data base (VM and interface).
    - The entry that rejected due to conflict (VM and interface).

## Mitigation

Inspect the duplicates.

> Remove or update mac addresses that were assigned manually or before kubemacpool was deployed or opted in.

Reviewer: ```suggestion
Update the VMs to remove the duplicate MAC Addresses.
```

> Once all the conflicts are resolved, restart the `kubemacpool-mac-controller` pod,

Reviewer: please add command example

> in order to rescan and rebuild the database.

Reviewer: I would remove this line, no further details needed imo.

# KubeMacPoolDuplicateMacsFound

## Meaning

KubeMacPool is responsible to allocate MAC Addresses and prevent MAC Address conflicts.
Upon starting KubeMacPool, it scans the cluster for MAC Addresses in VMs set on the managed namespaces.
In case it finds duplicates it will raise this alert.

## Impact

> Having a duplicate MAC Addresses can break the requirement that each interface on the same LAN,

Reviewer: can we rephrase this? something like 
```suggestion
Duplicate MAC Addresses on the same LAN may cause unexpected and various connectability issues in the network.
```

should have a unique MAC Address.

## Diagnosis

- Find the `kubemacpool-mac-controller` namespace and pod,

>   `kubectl get pod -A -l control-plane=mac-controller-manager --no-headers -o custom-columns=":metadata.namespace,:metadata.name"`

Reviewer: This line appears in the same line when looking in the `Display rich format` option, I think you need to add a new line. also in the rest
